src/magic.cpp

```diff
@@ -1070,11 +1070,18 @@ std::string spell::damage_type_string() const
     return name_by_dt( dmg_type() );
 }
 
+#ifndef __aarch64__
 // constants defined below are just for the formula to be used,
 // in order for the inverse formula to be equivalent
 constexpr float a = 6200.0f;
 constexpr float b = 0.146661f;
 constexpr float c = -62.5f;
+#else
+// ARM64 needs double
+constexpr double a = 6200.0f;
+constexpr double b = 0.146661f;
+constexpr double c = -62.5f;
```

Comment:
Shouldn't they lose the `f` here, since they're doubles, or is there magic (heh) I'm not seeing?

Reply:
i think they should but only because there's technically some implicit casting here i believe

Reply:
Oh right.

Reply:
Can we just use `exp_for_level` for the calculation of `spell::get_level` though? Subtract from the current exp until it is lower than the required exp of a level, and then return that level.

Reply:
> 
> 
> Can we just use `exp_for_level` for the calculation of `spell::get_level` though? Subtract from the current exp until it is lower than the required exp of a level, and then return that level.

This is actually more correct than just throwing more bits at the problem. Transcendental functions aren't that cheap to compute though. How about precomputing them in an array?

Reply:
Precomputing can indeed speed it up, and currently the experience for each level only depends on these constants so it should be pretty straighforward to implement too.

Reply:
I'm not a big fan of a precomputed table, but it might be the best option for another reason: we need to ensure that the thresholds for each level are the same across different platforms, because otherwise when you save your game on one platform and load it on another you might get inconsistent results.